PublicOnboardingDemo/clients/LandingPage/controllers/client/onStart.js

```diff
@@ -0,0 +1,32 @@
+    
+    var isPublic = client.isPublic();
+    
+    var intro = client.getWidget("WelcomeLine");
+    intro.text = isPublic ? "Welcome to VANTIQ! You are currently in Public Mode." : "Welcome to VANTIQ! You are currently in Private Mode.";
+    
+    var uuid = client.getUsername();
+    var username = client.getUserRecord();
+    var uuid_widget = client.getWidget("UUID");
+    var username_widget = client.getWidget("Username");
+    var pref_username = "";
+    
+    uuid_widget.text = "Your Unique User ID is: " + uuid;
+    
+    // at this point username returns null in some cases
+    // the following function checks every tenth of a second
+    // to try and request the username data - and also sets and displays
+    // the username of the current user
+    function getPrefUsername(){
+        var id = setInterval(check, 100);
+        function check() {
+            if (username !== null) {
+                clearInterval(id);
+                pref_username = username.preferredUsername;
+                username_widget.text = "Your Username is: " + 								pref_username;
+            } else {
+                username = client.getUserRecord();
+            }
```

Comment:
I think this interval loop could use a little adjusting. You currently grab the username on one iteration of the check function, but won't use it until the subsequent iteration. It would be better to just call getUserRecord at the start, and use the result and clear the interval immediately if it's non-null, otherwise let the interval trigger again.